olp-cpp-sdk-dataservice-read/src/generated/api/ResourcesApi.cpp

```diff
@@ -32,42 +32,34 @@
 namespace olp {
 namespace dataservice {
 namespace read {
+static const auto kCancelledErrorResponse = client::ApiError(
+            static_cast<int>(http::ErrorCode::CANCELLED_ERROR), "Operation Cancelled.");
 
-client::CancellationToken ResourcesApi::GetApis(
-    std::shared_ptr<client::OlpClient> client, const std::string& hrn,
-    const std::string& service, const std::string& service_version,
-    const ApisCallback& callback) {
-  return GetApis(*client, hrn, service, service_version, callback);
-}
-
-client::CancellationToken ResourcesApi::GetApis(
+ResourcesApi::ApisResponse ResourcesApi::GetApis(
     const client::OlpClient& client, const std::string& hrn,
     const std::string& service, const std::string& service_version,
-    const ApisCallback& callback) {
+    const client::CancellationContext& context) {
+  if(context.IsCancelled()){
+      return ApisResponse(kCancelledErrorResponse);
+  }
```

Comment:
Do we need it?

Reply:
Probably not like this but rather everything wrapped inside a context.ExecuteOrCancelled(...).

Reply:
ExecuteOrCancelled is already called in SendRequest